### PR TITLE
Pre commit

### PR DESCRIPTION
**Pull request by normanb**
Adds https://pre-commit.com/ and https://github.com/ambv/black hooks.

### CODE REVIEW

File: merge_rgba/__init__.py [hunk 1]
```diff
@@ -109,19 +110,22 @@ def merge_rgba_tool(sources, outtif, bounds=None, res=None, precision=7,
...
-                                boundless=True, masked=False)
+                temp = src.read(
+                    out=temp, window=src_window, boundless=True, masked=False
+                )
 
                 # pixels without data yet are available to write
                 write_region = np.logical_and(
-                    (dstarr[3] == 0),  # 0 is nodata
-                    (temp[3] != 0))
+                    (dstarr[3] == 0), (temp[3] != 0)  # 0 is nodata
```
Comment (perrygeo) [inline merge_rgba/__init__.py]: this black change is a little hard to swallow -  rearranging the code here could change the interpretation of inline comments.

File: .travis.yml [hunk 2]
```diff
@@ -17,6 +17,8 @@ install:
   - pip install -r requirements-dev.txt
   - pip install -e .
   - pip install coveralls
-script: "py.test --cov merge_rgba --cov-report term-missing"
+script:
+  - "py.test --cov merge_rgba --cov-report term-missing"
+  - "git diff-tree --no-commit-id --name-only -r $REVISION | xargs pre-commit run --files"
```
Comment (perrygeo) [inline .travis.yml]: pre-commit looks like exactly what we need - nice find!

File: merge_rgba/__init__.py [hunk 3]
```diff
@@ -5,11 +5,16 @@
 from rasterio.transform import Affine, rowcol
 
 
-logger = logging.getLogger('merge_rgba')
+logger = logging.getLogger(
+        "merge_rgba"
+        
+        
+        )
```
Comment (perrygeo) [inline merge_rgba/__init__.py]: I installed the `pre-commit` hook which caught and fixed the above change locally before the commit was pushed. 👍 

Then I removed the pre-commit hook, tried again and pushed the change - hoping that the `pre-commit` call in the Travis tests would catch it and exit with an error. Unfortunately, it looks like the tests passed anyways and the change snuck in anyhow. Not sure why, I need to look at the pre-commit cli more closely.

**Comment by normanb** [comment]
Fixed travis configuration to use last commit sha from a PR.

File: merge_rgba/__init__.py (hunk 1 above)
Comment (perrygeo) [reply merge_rgba/__init__.py]: This behavior has been noted as a bug: https://github.com/ambv/black/issues/282

**Pull request merged**